ROMFS/px4fmu_common/init.d/rc.mavlink

```diff
@@ -0,0 +1,341 @@
+			micrortps_client start -d /dev/rtps -b 921600 -l -1 -s 2000
+
+			set MAVLINK_DEV_0       "/dev/mavlink"
+			set MAVLINK_DEV_0_ARGS  "-m onboard -r 5000 -x"
+			set MAVLINK_DEV_0_BAUD  "921600"
+		fi
+	else
+		set MAVLINK_UART_2      "/dev/ttyS2"
+		set MAVLINK_UART_2_ARGS "-m onboard -r 80000 -x -f"
+		set MAVLINK_UART_2_BAUD "921600"
+	fi
+
+	# Disable mavlink start if the Benewake TFMini or LeddarOne are present.
+	if param greater SENS_EN_LEDDAR1 0 -o param greater SENS_EN_TFMINI 0
+	then
+	else
+		# Default MAVLINK stream on ttyS3.
+		set MAVLINK_UART_3      "/dev/ttyS3"
+		set MAVLINK_UART_3_ARGS "-r 1200"
+		set MAVLINK_UART_3_BAUD "57600"
+	fi
+fi
+
+if ver hwcmp NXPHLITE_V3
+then
+	# Enable MAVLINK_UART_4 on ttyS4 with default config.
+	set MAVLINK_UART_4 	"/dev/ttyS4"
+	set MAVLINK_UART_4_ARGS "-m osd -r 1000"
+	set MAVLINK_UART_4_BAUD "57600"
+fi
+
+if ver hwcmp MINDPX_V2
+then
+	# This is TELEM4 on Pixhawk 3 Pro.
+	frsky_telemetry start -d /dev/ttyS6
+fi
+
+if ver hwcmp PX4FMU_V4
+then
+	# This is TELEM4 on Pixhawk 3 Pro.
+	frsky_telemetry start -d /dev/ttyS6
+
+	# MAVLink on Wifi (ESP8266 port).
+	set MAVLINK_UART_0      "/dev/ttyS0"
+	set MAVLINK_UART_0_ARGS "-r 20000"
+	set MAVLINK_UART_0_BAUD "921600"
+
+	set MAVLINK_UART_1      "/dev/ttyS1"
+	set MAVLINK_UART_1_ARGS "-r 1200"
+	set MAVLINK_UART_1_BAUD "57600"
+fi
+
+if ver hwcmp PX4FMU_V4PRO
+then
+	# This is TELEM4 on Pixhawk 3 Pro.
+	frsky_telemetry start -d /dev/ttyS6
+fi
+
+if ver hwcmp PX4FMU_V5
+then
+	set MAVLINK_UART_1      "/dev/ttyS1"
+	set MAVLINK_UART_1_ARGS "-r 1200 -f"
+	set MAVLINK_UART_1_BAUD "57600"
+
+	set MAVLINK_UART_3      "/dev/ttyS3"
+	set MAVLINK_UART_3_ARGS "-r 20000"
+	set MAVLINK_UART_3_BAUD "57600"
+fi
+
+if ver hwcmp OMNIBUS_F4SD
+then
+	# Enable MAVLINK_UART_1 with default config.
+	set MAVLINK_UART_1      "/dev/ttyS1"
+
+	# Disable MAVLINK_UART_2 default.
+	set MAVLINK_UART_2      none
+fi
+
+###############################################################################
+#                 End Setup for board specific configurations.                #
+###############################################################################
+
+
+#
+# MAVLink onboard / TELEM2.
+#
+if param compare SYS_COMPANION 10
+then
+	frsky_telemetry start -d ${MAVLINK_DEV_2}
+fi
+
+if param compare SYS_COMPANION 20
+then
+	# TODO: this should now be relocated with a board specific configuration above.
+	set MAVLINK_DEV_0       "/dev/bridge0"
+	set MAVLINK_DEV_0_ARGS  "-m osd -r 40000"
+	set MAVLINK_DEV_0_BAUD  "57600"
+
+	syslink start
+fi
+
+#
+# 19200 Baud Rate.
+#
+if param compare SYS_COMPANION 319200
+then
+	set MAVLINK_UART_2_ARGS "-r 1000 -f"
+	set MAVLINK_UART_2_BAUD "19200"
+fi
+
+if param compare SYS_COMPANION 419200
+then
+	# TODO: this should now be relocated with a board specific configuration above.
+	set MAVLINK_DEV_0       "/dev/iridium"
+	set MAVLINK_DEV_0_ARGS  "-m iridium -r 10"
+	set MAVLINK_DEV_0_BAUD  "19200"
+
+	iridiumsbd start -d /dev/ttyS2
```

Comment:
This will conflict with the default of `MAVLINK_UART_2`. I think we should be using `MAVLINK_UART_2` here and drop `MAVLINK_DEV_0`. What do you think?

Also the USB mavlink instance does IMO not need to be configurable. This will reduce Flash usage further. You can also revert 2a3d66cf457e9b7879276bd01be7936876b56147.

Reply:
Thanks @bkueng,

I fixed up the MAVLINK_UART_2 conflict you identified, (thanks for spotting that), and I've removed the MAVLINK_USB_0 and MAVLINK_DEV_0 instances and reverted the commit you pointed out to save on flash.

I've reverted both the hardware specific cases at the top of rc.mavlink to simply start their respective mavlink instances to save on flash, however it still isn't enough.

Another ~2k of flash can be reclaimed by deprecating the usage of the `SYS_COMPANION` in rc.mavlink.  Airframe config files in this PR have been updated to allow this.  Let me know if this is acceptable to you.

Thanks!

-Mark

Reply:
Thanks for the update.

> Another ~2k of flash can be reclaimed by deprecating the usage of the SYS_COMPANION in rc.mavlink. Airframe config files in this PR have been updated to allow this. Let me know if this is acceptable to you.

My answer is still the same as in https://github.com/PX4/Firmware/pull/9065#issuecomment-390106087. I'd like to push for something like https://github.com/PX4/Firmware/pull/9615, then all UARTs are fully configurable via QGC with minimal flash overhead. The airframe configs can then also override the defaults.